### ISSUE TITLE
Kitty adds itself to path even when it exists in path

### ISSUE THREAD

**Issue by typkrft** [kovidgoyal/kitty#5472]
**Describe the bug**
Disclaimer:
This is probably not crucial at all, but it does add some clutter when I need to track things down in my path. Also forgive me as I am not an advanced developer by any means there may be implications I am not familiar with. 

Kitty intelligently adds itself to my `$PATH` hooray! But it was already there because I added it. This wouldn't really be an issue, I could just remove the path from my `.zshrc` but kitty adds it to the end of the path. Just for exmaple: In my situation that would mean the system python comes first then homebrew version of python I want to use. In practical use this has not ever effected me other than confusing me for a split second when I was trying to figure out where the extra path  was coming from. 

My `$PATH` in any other terminal
`/Users/typkrft/.config/bin:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin`

MY `$PATH` in kitty
`/Users/typkrft/.config/bin:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Applications/kitty.app/Contents/MacOS:/opt/homebrew/bin`

Notice:
`/opt/homebrew/bin` is added twice once by me and once by kitty. I don't know for sure but when kitty adds it self, but if it's before it sources my `.zshrc` then maybe this is unavoidable, or maybe this could be configurable. 

I've read quite a bit of the documentation over the years, but it is possible I missed something. I apologize if that's the case. I also couldn't quickly identify in the codebase where this occurs to try and change it myself. 

**To Reproduce**
Use kitty

**Screenshots**
NA

**Environment details**
```
kitty 0.26.1 created by Kovid Goyal
Darwin Typkrfts-MacBook-Pro.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
ProductName:	macOS ProductVersion:	12.4 BuildVersion:	21F79
Frozen: True
Paths:
  kitty: /Applications/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/kitty.app/Contents/Resources/Python/lib/kitty-extensions
  system shell: /bin/zsh
Loaded config files:
  /Users/typkrft/.config/kitty/kitty.conf

Config options different from defaults:
allow_remote_control     yes
cursor_shape             2
draw_minimal_borders     False
enable_audio_bell        False
enabled_layouts          ['fat:bias=50;full_size=1;mirrored=true', 'vertical', 'horizontal', 'grid', 'stack', 'splits:split_axis=horizontal', 'fat:bias=50;full_size=1;mirrored=false', 'tall:bias=50;full_size=1;mirrored=false']
focus_follows_mouse      True
font_family              VictorMono Nerd Font
font_size                18.0
hide_window_decorations  1
macos_colorspace         displayp3
macos_custom_beam_cursor True
macos_option_as_alt      3
modify_font:
    cell_height 140%
mouse_hide_wait          3.0
symbol_map:
	U+0030 - U+0039 → Iosevka Term Slab
tab_bar_edge             1
tab_switch_strategy      left
term                     xterm-256color
window_border_width      (2.5, 'pt')
window_padding_width     FloatEdges(left=4.0, top=0, right=4.0, bottom=0)
Added shortcuts:
	cmd+/ →  send_text all ♇
	cmd+a →  send_text all ♗
	cmd+backspace →  send_text all ☽
	cmd+left →  send_text all ☀︎
	cmd+right →  send_text all ☼
	cmd+z →  send_text all ☾
	shift+cmd+c →  send_text all ☃︎
	shift+cmd+left →  send_text all ☍
	shift+cmd+right →  send_text all ☌
	shift+cmd+z →  send_text all ☁︎
	shift+down →  move_window down
	shift+left →  move_window left
	shift+right →  move_window right
	shift+up →  move_window up
Changed shortcuts:
	cmd+down →  send_text all ☂︎
	cmd+k →  combine : clear_terminal scrollback active : send_text normal,application \x0c
	cmd+up →  send_text all ☔︎
	kitty_mod+down →  neighboring_window down
	kitty_mod+left →  neighboring_window left
	kitty_mod+right →  neighboring_window right
	kitty_mod+up →  neighboring_window up
Colors:
	active_border_color      #50fa7b   
	active_tab_background    #569fba   
	active_tab_foreground    #191726   
	background               #232136   
	color0                   #393552   
	color1                   #eb6f92   
	color10                  #b1d196   
	color11                  #f9cb8c   
	color12                  #65b1cd   
	color13                  #ccb1ed   
	color14                  #a6dae3   
	color15                  #e2e0f7   
	color16                  #ea9a97   
	color17                  #eb98c3   
	color2                   #a3be8c   
	color3                   #f6c177   
	color4                   #569fba   
	color5                   #c4a7e7   
	color6                   #9ccfd8   
	color7                   #e0def4   
	color8                   #47407d   
	color9                   #f083a2   
	cursor                   #e0def4   
	foreground               #e0def4   
	inactive_border_color    #44475a   
	inactive_tab_background  #433c59   
	inactive_tab_foreground  #817c9c   
	selection_background     #433c59   
	selection_foreground     #e0def4   
	url_color                #a3be8c   

Important environment variables seen by the kitty process:
	PATH                                /Applications/kitty.app/Contents/MacOS:/opt/homebrew/bin:/usr/bin:/bin:/usr/sbin:/sbin
	LANG                                en_US.UTF-8
	SHELL                               /bin/zsh
	USER                                typkrft

```

**Additional context**
Try to reproduce the problem with `kitty --config NONE` if you cannot then post a minimal kitty.conf that reproduces the problem. If the problem involves interaction with some other terminal program post a minimal config for that program to reproduce the problem as well.
This occurs using `--config NONE`.
Thanks!

**Comment by kovidgoyal** [comment]
kitty will only add itself to the PATH if the existing kitty in the path
is not the same kitty. Having the wrong kitty in the PATH will cause
many issues. See the code in ensure_kitty_in_path()

**Comment by typkrft** [comment]
Thank you for the explanation and the applicable code. 

If I go into another terminal and uninstall kitty `brew uninstall kitty` and then run `which -a kitty` I receive the expected `kitty not found`.  If I reinstall in the same browser `brew install --cask kitty` I get the follow:
```BASH
==> Downloading https://github.com/kovidgoyal/kitty/releases/download/v0.26.1/kitty-0.26.1.dmg
Already downloaded: /Users/typkrft/Library/Caches/Homebrew/downloads/36206a9e979833cb8e8db7f166d773600c85e9df519e787f55be4258506f91c6--kitty-0.26.1.dmg
==> Installing Cask kitty
==> Moving App 'kitty.app' to '/Applications/kitty.app'
==> Linking Binary 'kitty.wrapper.sh' to '/opt/homebrew/bin/kitty'
🍺  kitty was successfully installed!
```
If I run `which -a kitty` again I get brew's linked binary `/opt/homebrew/bin/kitty`.  If I open kitty with brew's linked binary I get the behavior described above. If I open kitty.app. I do not get the duplicate entry. This is odd because the linked wrapper binary just points to the kitty.app. If I try to open kitty from the `Kitty.app` with `kitty` which is currently `/opt/homebrew/bin/kitty` then I get `/Users/brandon/.config/bin:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Users/brandon/.config/bin:/opt/homebrew/bin:/Applications/kitty.app/Contents/MacOS`. Which has a duplicate similar to above but also seems to reverse the order where as the homebrew path comes first and the .app path comes second. 

I'm not going to reopen this because it could just be something I'm doing wrong or I'm still not understanding and it' really hasn't caused any problems as of yet.

TLDR:
1. Kitty not installed 
- `which -a kitty`: `kitty not found`
- `echo $PATH`: `/Users/typkrft/.config/bin:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin`
2. Kitty installed via brew opened via /Applications/kitty.app 
- from kitty `which -a kitty`: `/opt/homebrew/bin/kitty /Applications/kitty.app/Contents/MacOS/kitty`
- from terminal.app `which -a kitty`: `/opt/homebrew/bin/kitty`
- `echo $PATH`:  `/Users/typkrft/.config/bin:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/Applications/kitty.app/Contents/MacOS`
3. Kitty opened via shell 
- from kitty `which -a kitty`:
```SHELL
/opt/homebrew/bin/kitty
/Applications/kitty.app/Contents/MacOS/kitty
/opt/homebrew/bin/kitty
```
- from terminal.app `which -a kitty`: `/opt/homebrew/bin/kitty`

**Comment by kovidgoyal** [comment]
The cpde checks for symlinks, so presumably homebrew's wrapper binary is
not a symlink but an actual binary. There is no way for kitty to know
what that binary is or whether it is the correct binary. homebrew should
be updated to use a symlink and you will no longer have the issue. Or
just use the official kitty binary instead of homebrew.

**Comment by typkrft** [comment]
Installing the latest stable and nightly from curl both correctly show only 1 `kitty` via `which -a kitty`, which is `/Applications/kitty.app/Contents/MacOS/kitty`. Running from another terminal gives the expected `kitty not found`. So all is good there. However it still adds duplicate entries to $PATH. 

Brew must be doing something I assume though I can only replicate with kitty. Running `/Applications/kitty.app/Contents/MacOS/kitty --config NONE /bin/bash` No bashrc. No duplicates. Do the same with zsh results in no duplicates if I don't use a zshrc. The weird thing is that even though this doesn't happen anywhere else the `/opt/homebrew/bin/` folder still immediately proceeds `/Applications/kitty.app/Contents/MacOS`. No zshrc in terminal.app and iTerm results in `/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin`

**Comment by bobbykjack** [comment]
@Ducky6944 Not sure if this is still an issue for you 3 years on...

I found myself going down the same rabbit hole today. Looking at the code for ensure_kitty_in_path and trying to work out how to suppress it.

For now, I've given up and I've added the following to my ~/.zshrc:

`PATH=$(echo "$PATH" | sed -e 's/:\/Applications\/kitty.app\/Contents\/MacOS$//')`

This simply manually removes the path from PATH: a pretty poor solution, but it's the only thing I've managed to get to work!
